data-prepper-core/src/test/java/org/opensearch/dataprepper/parser/PipelineParserTests.java

```diff
@@ -98,29 +122,48 @@ void parseConfiguration_with_missing_pipeline_name_should_throw() {
 
     @Test
     void parseConfiguration_with_missing_pipeline_name_in_multiple_pipelines_should_throw() {
-        final PipelineParser pipelineParser = new PipelineParser(TestDataProvider.MISSING_PIPELINE_MULTIPLE_PIPELINE_CONFIG_FILE, pluginFactory, peerForwarderProvider);
+        final PipelineParser pipelineParser =
+                new PipelineParser(TestDataProvider.MISSING_PIPELINE_MULTIPLE_PIPELINE_CONFIG_FILE, pluginFactory, peerForwarderProvider, dataPrepperConfiguration);
         final RuntimeException actualException = assertThrows(RuntimeException.class, pipelineParser::parseConfiguration);
         assertThat(actualException.getMessage(), equalTo("Invalid configuration, no pipeline is defined with name test-pipeline-4"));
     }
 
     @Test
     void testMultipleSinks() {
-        final PipelineParser pipelineParser = new PipelineParser(TestDataProvider.VALID_MULTIPLE_SINKS_CONFIG_FILE, pluginFactory, peerForwarderProvider);
+        mockDataPrepperConfigurationAccesses();
+        final PipelineParser pipelineParser =
+                new PipelineParser(TestDataProvider.VALID_MULTIPLE_SINKS_CONFIG_FILE, pluginFactory, peerForwarderProvider, dataPrepperConfiguration);
         final Map<String, Pipeline> pipelineMap = pipelineParser.parseConfiguration();
         assertThat(pipelineMap.keySet().size(), equalTo(3));
+        verifyDataPrepperConfigurationAccesses(pipelineMap.keySet().size());
     }
 
     @Test
     void testMultipleProcessors() {
-        final PipelineParser pipelineParser = new PipelineParser(TestDataProvider.VALID_MULTIPLE_PROCESSERS_CONFIG_FILE, pluginFactory, peerForwarderProvider);
+        mockDataPrepperConfigurationAccesses();
+        final PipelineParser pipelineParser =
+                new PipelineParser(TestDataProvider.VALID_MULTIPLE_PROCESSERS_CONFIG_FILE, pluginFactory, peerForwarderProvider, dataPrepperConfiguration);
         final Map<String, Pipeline> pipelineMap = pipelineParser.parseConfiguration();
         assertThat(pipelineMap.keySet().size(), equalTo(3));
+        verifyDataPrepperConfigurationAccesses(pipelineMap.keySet().size());
     }
 
     @Test
     void parseConfiguration_with_a_configuration_file_which_does_not_exist_should_throw() {
-        final PipelineParser pipelineParser = new PipelineParser("file_does_no_exist.yml", pluginFactory, peerForwarderProvider);
+        final PipelineParser pipelineParser = new PipelineParser("file_does_no_exist.yml", pluginFactory, peerForwarderProvider, dataPrepperConfiguration);
         final RuntimeException actualException = assertThrows(RuntimeException.class, pipelineParser::parseConfiguration);
         assertThat(actualException.getMessage(), equalTo("Failed to parse the configuration file file_does_no_exist.yml"));
     }
+
+    private void mockDataPrepperConfigurationAccesses() {
+        when(dataPrepperConfiguration.getProcessorShutdownTimeout()).thenReturn(Duration.ofSeconds(new Random().nextInt()));
```

Comment:
Thanks for using random int! Are negative numbers going to cause any issues here? Maybe add an upper bound or use `abs`?

Reply:
Good catch, I'll add `abs`